protocol-designer/src/labware-ingred/types.ts

```diff
@@ -19,23 +19,21 @@ export interface WellContents {
   selected?: boolean
   maxVolume?: number
 }
-export type ContentsByWell = {
-  [wellName: string]: WellContents
-} | null
-export interface WellContentsByLabware {
-  [labwareId: string]: ContentsByWell
-}
+export type ContentsByWell = Record<string, WellContents> | null
+export type WellContentsByLabware = Record<string, ContentsByWell>
 // ==== INGREDIENTS ====
 export type OrderedLiquids = Array<{
   ingredientId: string
   name: string | null | undefined
   displayColor: string | null | undefined
+  liquidClass: string | null | undefined
```

Comment:
nit, the other fields can be refactored as well. 
 
```suggestion
  liquidClass?: string | null
```

Reply:
is this undefined here to avoid needing to add this to a migration?

Reply:
I think these are unnecessarily allowed to be undefined or null in some cases. The modal disallows saving without a liquid name, and displayColor is always a valid hex string. I think description can be set to null for easier use downstream as well. Thoughts on:
```
export interface LiquidGroup {
  name: string
  description: string | null
  displayColor: string
  liquidClass: string | null
  serialize: boolean
}

```

Reply:
and since we're dealing with nullish initial values in DefineLiquidsModal, we can update that interface to 
```
interface LiquidEditFormValues {
  name: string
  displayColor: string
  description: string
  liquidClass: string
  serialize: boolean
  [key: string]: unknown
}
```

Reply:
i see, that makes sense to me. 

i think we would need `liquidClass` to be undefined in `OrderedLiquids` at least because its included in the JSON protocols and if it is always null or a string, we'd need to make a migration for it. i'm all for adding this to the migration but not right now since there's no way to put it behind a FF. Do you mind adding a TODO or something above the type to remind us to migrate it later?

Reply:
For sure! I will keep it as an optional property for now and we can remove the optionality and add a migration later